src/datachain/toolkit/split.py

```diff
@@ -58,14 +69,16 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
 
     weights_normalized = [weight / sum(weights) for weight in weights]
 
-    resolution = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
+    rand_col = C("sys.rand")
+    if seed is not None:
+        uniform_seed = random.Random(seed).randrange(1, RESOLUTION)  # noqa: S311
+        rand_col = (rand_col % RESOLUTION) * uniform_seed  # type: ignore[assignment]
```

Comment:
`(rand_col % RESOLUTION)` is needed because of [this](https://github.com/iterative/datachain/pull/657#issuecomment-2530115157):

Python:

```
In [1]: 8822966792743105186 * 1730071088
Out[1]: 15264359758508934493641462368

In [2]: (8822966792743105186 * 1730071088) % 2147483647
Out[2]: 581439120
```

SQLite:

```
sqlite> select 8822966792743105186 * 1730071088;
1.52643597585089e+28

sqlite> select (8822966792743105186 * 1730071088) % 2147483647;
1.0
```

Reply:
My bad, thanks! I assumed that SQLite was displaying "scientific notation" but it wouldn't be an issue. 🙈

Reply:
On _modulo_ operations, `RESOLUTION` should be `RESOLUTION + 1` as well.

```suggestion
        rand_col = (rand_col % (RESOLUTION + 1)) * uniform_seed  # type: ignore[assignment]
```